runtime/alloc.c

```diff
@@ -83,70 +76,77 @@ Caml_inline value do_alloc_small(mlsize_t wosize, tag_t tag, value* vals)
     CAMLdrop;                                               \
   } while (0)
 
-  Alloc_small(v, wosize, tag, Enter_gc);
-  for (i = 0; i < wosize; i++) {
-    Field(v, i) = vals[i];
-  }
-  return v;
+#define DO_ALLOC_SMALL(wosize, tag, vals) \
```

Comment:
Nit-pick. I'd write `Do_alloc_small` and use parentheses around the occurrences of the arguments, e.g. `(tag) < 256`.

Can you please one comment line here to explain to future readers why you had to use a macro?